src/test/java/org/datadog/jmxfetch/TestReconnectContainer.java

```diff
@@ -47,24 +57,47 @@ private static boolean isDomainPresent(String domain, MBeanServerConnection mbs)
     private static ImageFromDockerfile img = new ImageFromDockerfile()
         .withFileFromPath(".", Paths.get("./tools/misbehaving-jmx-server/"));
 
-    @Before
-    public void setup() {
-        this.controlClient = new JMXServerControlClient(cont.getHost(), cont.getMappedPort(controlPort));
-    }
 
-    @Rule
+    @Rule(order = 0)
     public GenericContainer<?> cont = new GenericContainer<>(img)
-        .withExposedPorts(rmiPort, controlPort)
+        .withExposedPorts(rmiPort, controlPort, supervisorPort)
         .withEnv(Collections.singletonMap("RMI_PORT", "" + rmiPort))
         .withEnv(Collections.singletonMap("CONTROL_PORT", "" + controlPort))
-        .waitingFor(Wait.forLogMessage(".*IAMREADY.*", 1));
+        .withEnv(Collections.singletonMap("SUPERVISOR_PORT", "" + supervisorPort))
+        .waitingFor(Wait.forLogMessage(".*JMX server has started.*", 1));
+        // TODO switch to a http ready check, but need to be able to address container by container IP, not gateway/docker-host
```

Comment:
address this, if its possible then do it, otherwise double check that the log will not be triggered before the JMX server is actually ready.